packages/orm-integration/tests/integration/relation-bug.spec.ts

```diff
@@ -0,0 +1,90 @@
+import 'reflect-metadata';
+import { entity, getClassSchema, plainToClass, t } from '@deepkit/type';
+import { test } from '@jest/globals';
+import { createEnvSetup } from './setup';
+
+test('relation bug spec', async () => {
+    const cookieSchema = t.schema({
+        id: t.number.autoIncrement.primary,
+        type: t.string,
+    }, { name: 'relationBugCookie' });
+
+    const userSchema = t.schema({
+        id: t.number.autoIncrement.primary,
+        username: t.string,
+        cookies: t.array(cookieSchema).optional,
```

Comment:
with `.reference()` was the use-case, right? Only properties marked as `reference` or `backReference` can be used in `join` methods.